File: src/modules/categories/entities/Category.ts
import { v4 as uuidV4 } from "uuid";
import { Column, CreateDateColumn, Entity, ManyToOne, PrimaryColumn, UpdateDateColumn } from "typeorm";
import { User } from "../../accounts/entities/User";

@Entity("categories")
class Category {
    @PrimaryColumn()
    id?: string;

    @Column()
    name: string;

    @Column()
    type: string;

    @ManyToOne(() => User, (user) => user.categories)
    user_id = User

    @CreateDateColumn()
    created_at?: Date;

    @UpdateDateColumn()
    updated_at?: Date;

    @CreateDateColumn()
    excluded_at?: Date;

    constructor() {
        if(!this.id){
            this.id = uuidV4();
        }

        this.name = "";
        this.type = "";
    }
}

export { Category };

File: src/main/java/br/com/ottonsam/gastanaoapi/entity/Category.java
package br.com.ottonsam.gastanaoapi.entity;

import br.com.ottonsam.gastanaoapi.entity.dtos.CreateCategoryDto;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.UUID;

@Entity
@Table(name = "categories")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Category {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;

    @Column(nullable = false)
    private String description;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    public Category(CreateCategoryDto expenseDto, User user) {
        this.description = expenseDto.description();
        this.user = user;
    }
}
